v3-docs/v3-migration-docs/breaking-changes/call-x-callAsync.md

```diff
@@ -32,105 +28,209 @@ Meteor.call('getAllData', (err, res) => { // [!code highlight]
+import { Meteor } from "meteor/meteor";
+import { Mongo } from "meteor/mongo";
 
-const MyCollection = new Mongo.Collection('myCollection');
+const MyCollection = new Mongo.Collection("myCollection");
 
 Meteor.methods({
   async getAllData() {
     return await MyCollection.find().fetchAsync(); // [!code highlight]
-  }
+  },
 });
-
 ```
 
 :::
 
 ## Rules of using `Meteor.callAsync` & `Meteor.call`
 
-::: tip 
-It is not recommended to use concurrent calls.
-Use `await` for your `Meteor.callAsync`.
-:::
+You can still use `Meteor.callAsync` and `Meteor.call` together. But there are some catches that you need to
+pay attention to. Let's go over them.
+
+### How to NOT use them together
 
-here are a few examples of cases where you should use `Meteor.callAsync` instead of `Meteor.call`:
+These are for **async** method when called with `Meteor.callAsync` or `Meteor.call`.
+
+**Case 1**: Calling `Meteor.callAsync` and then right after calling `Meteor.call` when the methods have stubs.
 
 ```js
-import { Meteor } from 'meteor/meteor'
+import { Meteor } from "meteor/meteor";
+
+Meteor.callAsync("SOME_ASYNC_METHOD_WITH_STUB")
+  .then((data) => console.log(data))
+  .catch((err) => console.log(err));
 
-Meteor.call("someMethod", (err, res) => { // [!code error] This is not ok
+Meteor.call("SOME_ASYNC_METHOD_WITH_STUB", (err, res) => {
   if (err) {
     console.log(err);
   } else {
     console.log(res);
   }
 });
+```
 
+**Case 2**: Calling `Meteor.call` and then right after calling `Meteor.callAsync` when the methods have stubs.
 
-Meteor.callAsync('someMethod') // [!code error] This is not ok
-  .then(data => console.log(data))
-  .catch(err => console.log(err));
+```js
+import { Meteor } from "meteor/meteor";
 
-// it is not recommended to use concurrent calls
+Meteor.call("SOME_ASYNC_METHOD_WITH_STUB", (err, res) => {
+  if (err) {
+    console.log(err);
+  } else {
+    console.log(res);
+  }
+});
 
-Promise.all([  // [!code error] This is not ok
-  Meteor.callAsync('someMethod'),
-  Meteor.callAsync('someMethod')
-]).then(([data1, data2]) => {
-  console.log(data1, data2);
+Meteor.callAsync("SOME_ASYNC_METHOD_WITH_STUB")
+  .then((data) => console.log(data))
+  .catch((err) => console.log(err));
+```
+
+**Case 3**: Calling `Meteor.callAsync` and then right after calling `Meteor.call` but the method called with `Meteor.call`
+does not have stub.
+
+```js
+import { Meteor } from "meteor/meteor";
+
+Meteor.callAsync("SOME_ASYNC_METHOD_WITH_STUB")
+  .then((data) => console.log(data))
+  .catch((err) => console.log(err));
+
+Meteor.call("SOME_ASYNC_METHOD_WITHOUT_STUB", (err, res) => {
+  if (err) {
+    console.log(err);
+  } else {
+    console.log(res);
+  }
+});
+```
+
+**Case 4**: Calling `Meteor.call` and then right after calling `Meteor.callAsync` but the method called with `Meteor.call`
+does not have stub.
+
+```js
+import { Meteor } from "meteor/meteor";
+
+Meteor.call("SOME_ASYNC_METHOD_WITHOUT_STUB", (err, res) => {
```

Comment:
I think using `ASYNC_METHOD` makes this a little confusing. The client isn't affected by if the method is async or not on the server. It might be better to instead say if the stub is async or sync, since that is what affects the client.